Tests/SuperDelegate+RemoteNotificationsTests.swift

```diff
@@ -42,7 +42,7 @@ class SuperDelegateRemoteNotificationTests: SuperDelegateTests {
         let remoteNotificationCapableDelegate = RemoteNotificationCapableDelegate()
         let registrationError = NSError(domain: "push registration test", code: 0, userInfo: nil)
         remoteNotificationCapableDelegate.application(UIApplication.shared, didFailToRegisterForRemoteNotificationsWithError: registrationError)
-        if let registeredFailure = remoteNotificationCapableDelegate.registerFailure as? NSError {
+        if let registeredFailure = remoteNotificationCapableDelegate.registerFailure as NSError? {
```

Comment:
I'm curious what caused this change? I believe these two lines are equivalent. If this is required by Swift 4, can you let me know what error the prior version triggered? This is a surprising change to me :)

Reply:
![image](https://user-images.githubusercontent.com/5439810/34263935-72dc9502-e671-11e7-88f2-818c546eff9d.png)

Mainly because of that Xcode hint.
The reason for that is the following:

Since Swift 4, there exist types which can be converted between Objective-C (`NSError`) and Swift (`Error`). These so called bridging types always succeeded. Thus, we only would need to change:

```swift
if let registeredFailure = remoteNotificationCapableDelegate.registerFailure as? NSError {}
```

into this:

```swift
if let registeredFailure = remoteNotificationCapableDelegate.registerFailure as NSError {}
```

But as `registerFailure` in `RemoteNotificationCapableDelegate` is an optional `Error?`, we need to cast it to an optional as well.

Reply:
Fascinating! Thanks for letting me know 😄.